src/main/java/edu/cornell/kfs/fp/document/validation/impl/CuDisbursementVoucherDocumentPreRules.java

```diff
@@ -64,4 +96,151 @@ protected boolean checkWireTransferTabState(DisbursementVoucherDocument dvDocume
         return tabStatesOK;
     }
 
+    /**
+     * This method notifies the user if the Check Stub Text will break into more lines than those
+     * allowed in the FastTrack file format and optionally allows the user to go back and fix the issue.
+     * If the user chooses to fix the issue, it will return false, otherwise true.
+     *
+     * @param dvDocument submitted disbursement voucher document
+     * @return Returns true if the user wishes to correct the Check Stub Text, false otherwise.
+     */
+    @SuppressWarnings("deprecation")
+    protected boolean checkCheckStubTextOverflow(DisbursementVoucherDocument dvDocument) {
+        boolean checkStubTextOK = true;
+        int maxCheckStubLines = CuDisbursementVoucherDocumentPreRules.FASTTRACK_MAX_LINE_COUNT;
+
+        List<PaymentNoteText> pntList = buildNotesForCheckStubText(dvDocument.getDisbVchrCheckStubText());
+
+        if(pntList.size() > maxCheckStubLines) {
+            String questionText = SpringContext.getBean(ConfigurationService.class).getPropertyValueAsString(CUKFSKeyConstants.QUESTION_OVERSIZED_CHECK_STUB_TEXT);
+            boolean useOversizedCheckStubText = super.askOrAnalyzeYesNoQuestion(CuFPConstants.USE_OVERSIZED_CHECK_STUB_TEXT_QUESTION_ID, questionText);
+
+            if(!useOversizedCheckStubText) {
+                // return to document if the user doesn't want to use the oversized Check Stub Text
+                super.event.setActionForwardName(KFSConstants.MAPPING_BASIC);
+                checkStubTextOK = false;
+            }
+
+        }
+
+        return checkStubTextOK;
+    }
+
+    /* This is borrowed mostly from
+     * edu.cornell.kfs.document.service.impl.CuDisbursementVoucherExtractionHelperServiceImpl#buildNoteForCheckStubText
+     * with the modification of removing the setting for the previous line count (as we don't have previous lines, and
+     * we're really only interested in how the existing Check Stub Text lines will be broken up).
+     */
+    private List<PaymentNoteText> buildNotesForCheckStubText(String text) {
+        if(text == null) {
+            throw new IllegalArgumentException("Cannot build notes for the Check Stub text from a null string.");
```

Comment:
I would advise checking if this is indeed the correct way to handle the null text argument, given that the caller is retrieving it from the DV document. If other checks already exist elsewhere to prevent submission of a DV with null check stub text, then I would advise doing some other handling here (like returning an empty list) if throwing the IllegalArgumentException ends up showing an incident report screen to the user.